llama-index-integrations/multi_modal_llms/llama-index-multi-modal-llms-nvidia/llama_index/multi_modal_llms/nvidia/base.py

```diff
@@ -0,0 +1,384 @@
+        if usage is None:
+            return {}
+
+        return {
+            "prompt_tokens": usage.get("prompt_tokens", 0),
+            "completion_tokens": usage.get("completion_tokens", 0),
+            "total_tokens": usage.get("total_tokens", 0),
+        }
+
+    def _complete(
+        self, prompt: str, image_documents: Sequence[ImageNode], **kwargs: Any
+    ) -> CompletionResponse:
+        all_kwargs = self._get_model_kwargs(**kwargs)
+        message_dict, extra_headers = self._get_multi_modal_chat_messages(
+            prompt=prompt, role=MessageRole.USER, image_documents=image_documents
+        )
+
+        response = self._client.request(
+            endpoint=NVIDIA_MULTI_MODAL_MODELS[self.model]["endpoint"],
+            stream=False,
+            messages=message_dict,
+            extra_headers=extra_headers,
+            **all_kwargs,
+        )
+        response = response.json()
+        text = response["choices"][0]["message"]["content"]
+        return CompletionResponse(
+            text=text,
+            raw=response,
+            additional_kwargs=self._get_response_token_counts(response),
+        )
+
+    def _stream_complete(
+        self, prompt: str, image_documents: Sequence[ImageNode], **kwargs: Any
+    ) -> CompletionResponseGen:
+        all_kwargs = self._get_model_kwargs(**kwargs)
+        message_dict, extra_headers = self._get_multi_modal_chat_messages(
+            prompt=prompt, role=MessageRole.USER, image_documents=image_documents
+        )
+
+        def gen() -> CompletionResponseGen:
+            response = self._client.request(
+                messages=message_dict,
+                stream=True,
+                endpoint=NVIDIA_MULTI_MODAL_MODELS[self.model]["endpoint"],
+                extra_headers=extra_headers,
+                **all_kwargs,
+            )
+            for line in response.iter_lines():
+                if line and line.strip() != b"data: [DONE]":
+                    line = line.decode("utf-8")
+                    line = line[5:]
+
+                    msg, final_line = aggregate_msgs(process_response(line))
+
+                    yield CompletionResponse(
+                        **msg,
+                        additional_kwargs=self._get_response_token_counts(line),
+                    )
+
+                    if final_line:
+                        break
+
+        return gen()
+
+    def complete(
+        self, prompt: str, image_documents: Sequence[ImageNode], **kwargs: Any
+    ) -> CompletionResponse:
+        return self._complete(prompt, image_documents, **kwargs)
+
+    def stream_complete(
+        self, prompt: str, image_documents: Sequence[ImageNode], **kwargs: Any
+    ) -> CompletionResponseGen:
+        return self._stream_complete(prompt, image_documents, **kwargs)
+
+    def chat(
+        self,
+        **kwargs: Any,
+    ) -> Any:
+        raise NotImplementedError("This function is not yet implemented.")
```

Comment:
please implement.

suggestion: rename _complete -> _chat, use generate_nvidia_multi_modal_chat_message + _chat in complete